x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -303,97 +304,71 @@ public void analyzedPlan(
-                    // CCS indices and skip_unavailable cluster values can stop the analysis right here
-                    if (analyzeCCSIndices(executionInfo, targetClusters, unresolvedPolicies, listenerResult, logicalPlanListener, l))
-                        return;
-                }
-                // whatever tuple we have here (from CCS-special handling or from the original pre-analysis), pass it on to the next step
-                l.onResponse(listenerResult);
-            })
-            .<ListenerResult>andThen((l, listenerResult) -> {
-                // first attempt (maybe the only one) at analyzing the plan
-                analyzeAndMaybeRetry(analyzeAction, requestFilter, listenerResult, logicalPlanListener, l);
-            })
-            .<ListenerResult>andThen((l, listenerResult) -> {
-                assert requestFilter != null : "The second pre-analysis shouldn't take place when there is no index filter in the request";
-
-                // "reset" execution information for all ccs or non-ccs (local) clusters, since we are performing the indices
-                // resolving one more time (the first attempt failed and the query had a filter)
-                for (String clusterAlias : executionInfo.clusterAliases()) {
-                    executionInfo.swapCluster(clusterAlias, (k, v) -> null);
-                }
-
-                // here the requestFilter is set to null, performing the pre-analysis after the first step failed
-                preAnalyzeIndices(preAnalysis.indices, executionInfo, listenerResult, null, l);
-            })
-            .<LogicalPlan>andThen((l, listenerResult) -> {
-                assert requestFilter != null : "The second analysis shouldn't take place when there is no index filter in the request";
-                LOGGER.debug("Analyzing the plan (second attempt, without filter)");
-                LogicalPlan plan;
-                try {
-                    plan = analyzeAction.apply(listenerResult.indices, listenerResult.lookupIndices, listenerResult.enrichResolution);
-                } catch (Exception e) {
-                    l.onFailure(e);
-                    return;
-                }
-                LOGGER.debug("Analyzed plan (second attempt, without filter):\n{}", plan);
-                l.onResponse(plan);
-            })
-            .addListener(logicalPlanListener);
-    }
+        var listener = SubscribableListener.<EnrichResolution>newForked(
+            l -> enrichPolicyResolver.resolvePolicies(targetClusters, unresolvedPolicies, l)
+        ).<ListenerResult>andThen((l, enrichResolution) -> resolveFieldNames(parsed, enrichResolution, l));
+        // first resolve the lookup indices, then the main indices
+        // if (indices.size() > 1) {
+        // // Note: JOINs on more than one index are not yet supported
+        // listener.onFailure(new MappingException("More than one LOOKUP JOIN is not supported"));
```

Comment:
Leftover?

Actually, I thought we had a check like this here a while ago. `preAnalysis.indices.size() <= 1` is still something we should assert, no? There can only be 1 main index (pattern) atm.

Reply:
Actually I put this here temporarily as a fallback option if we, for some other reason, wanted to revert to single-index lookups. This is now the easiest place to do that check. My plan was, once all tests pass, remove this code. So I think it is now time to do that.

Reply:
This came from the check in preAnalyzeLookupIndices, which no longer does that check. However, the preAnalyzeIndices method still does this check, and that is about multiple FROM commands. I think it likely that any future JOIN work will happen on top of lookup-join, so the check in preAnalyzeIndices should probably stay forever.